zio-http/src/main/scala/zhttp/service/server/content/handlers/ServerResponseHandler.scala

```diff
@@ -0,0 +1,125 @@
+package zhttp.service.server.content.handlers
+
+import io.netty.buffer.{ByteBuf, Unpooled}
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.{ChannelHandlerContext, DefaultFileRegion, SimpleChannelInboundHandler}
+import io.netty.handler.codec.http._
+import zhttp.core.Util
+import zhttp.http.{HTTP_CHARSET, HttpData, Response, Status}
+import zhttp.service.server.ServerTimeGenerator
+import zhttp.service.{ChannelFuture, HttpRuntime, Server}
+import zio.stream.ZStream
+import zio.{UIO, ZIO}
+
+import java.io.File
+
+@Sharable
+private[zhttp] case class ServerResponseHandler[R](
+  runtime: HttpRuntime[R],
+  config: Server.Config[R, Throwable],
+  serverTime: ServerTimeGenerator,
+) extends SimpleChannelInboundHandler[Response](false) {
+
+  type Ctx = ChannelHandlerContext
+
+  override def channelRead0(ctx: Ctx, response: Response): Unit = {
+    implicit val iCtx: ChannelHandlerContext = ctx
+
+    response.status match {
```

Comment:
why do we need to check for `Status`? Can't we simply encode and write to the channel, like we were doing before?

Reply:
simplified the code by avoiding pattern matching on status